src/teleop/jetson/jetson_teleop.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# Node for teleop-related callback functions
+
+from math import copysign
+from enum import IntEnum
+import rospy as ros
+from sensor_msgs.msg import Joy, JointState
+from geometry_msgs.msg import Twist
+from mrover.msg import Chassis, RAOpenLoopCmd, HandCmd
+
+
+def quadratic(val):
+    return copysign(val**2, val)
+
+
+def deadzone(magnitude, threshold):
+    temp_mag = abs(magnitude)
+    if temp_mag <= threshold:
+        temp_mag = 0
+    else:
+        temp_mag = (temp_mag - threshold) / (1 - threshold)
+
+    return copysign(temp_mag, magnitude)
+
+
+class Drive:
+    class LogitechAxes(IntEnum):
+        left_right = 0
+        forward_back = 1
+        twist = 2
+        dampen = 3
+        pan = 4
+        tilt = 5
+
+    # Constants for diff drive
+    TRACK_RADIUS = 10.0  # meter
+    WHEEL_RADIUS = 0.5  # meter
+
+    drive_vel_pub = ros.Publisher("/drive_cmd_wheels", Chassis, queue_size=100)
+
+    # TODO: Reimplement Dampen Switch
+    def teleop_drive_callback(self, msg):
+
+        # teleop_twist_joy angular message is maxed at 0.5 regardless of
+        # turbo mode, multiply by 2 to make range [-1,1]
+        v = msg.linear.x
+        omega = msg.angular.z * 2
+
+        # Transform into L & R wheel angular velocities using diff drive kinematics
+        omega_l = (v - omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+        omega_r = (v + omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+
+        command = Chassis()
+        command.omega_l = omega_l
+        command.omega_r = omega_r
+
+        self.drive_vel_pub.publish(command)
+
+
+class ArmControl:
+    class XboxMappings(IntEnum):
+        left_js_x = 0
+        left_js_y = 1
+        left_trigger = 6
+        right_trigger = 7
+        right_js_x = 2
+        right_js_y = 3
+        right_bumper = 5
+        left_bumper = 4
+        d_pad_up = 12
+        d_pad_down = 13
+        d_pad_right = 14
+        d_pad_left = 15
+        a = 0
+        b = 1
+        x = 2
+        y = 3
+
+    # RA Joint Publishers
+    joint_a_pub = ros.Publisher("/ra/open_loop/joint_a", JointState, queue_size=100)
+    joint_b_pub = ros.Publisher("/ra/open_loop/joint_b", JointState, queue_size=100)
```

Comment:
We can probably change the queue sizes to 1 instead of 100. I can't think of any reason for why we would want anything but the most recent command.

Ref for why we would maybe stick with a larger queue size [here](https://stackoverflow.com/questions/56444248/reason-to-set-queue-size-of-ros-publisher-or-subscriver-to-a-large-value)

Reply:
I chose it kinda arbitrarily base don code I saw online I'm also not really sure what the ideal size would be

Reply:
kk all good